components/hab/src/license.rs

```diff
@@ -0,0 +1,122 @@
+                    users::get_current_username}};
+use chrono::prelude::*;
+use dirs;
+use serde_yaml;
+use std::{fs::{self,
+               File},
+          io::Write,
+          path::PathBuf};
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct LicenseData {
+    pub date_accepted: DateTime<Utc>,
+    pub accepting_product: String,
+    pub accepting_product_version: String,
+    pub user: Option<String>,
+    pub file_format: String,
+}
+
+impl LicenseData {
+    pub fn new() -> Self {
+        LicenseData { date_accepted: Utc::now(),
+                      accepting_product: String::from("hab"),
+                      accepting_product_version: super::VERSION.to_string(),
+                      user: get_current_username(),
+                      file_format: String::from("1.0"), }
+    }
+}
+
+pub fn check_for_license_acceptance_and_prompt(ui: &mut UI) -> Result<()> {
+    if license_exists() {
+        return Ok(());
+    }
+
+    ui.heading("+---------------------------------------------+")?;
+    ui.heading("            Chef License Acceptance")?;
+    ui.br()?;
+    ui.info("Before you can continue, 1 product license must be accepted.")?;
+    ui.info("View the license at https://www.chef.io/end-user-license-agreement")?;
+    ui.br()?;
+    ui.info("License that needs accepting:")?;
+    ui.br()?;
+    ui.info("  * Habitat")?;
+    ui.br()?;
+
+    if ui.prompt_yes_no("Do you accept the 1 product license?", Some(false))? {
+        accept_license(ui)
+    } else {
+        ui.br()?;
+        ui.info("If you do not accept this license you will not be able to use Chef products.")?;
+        ui.br()?;
+
+        if ui.prompt_yes_no("Do you accept the 1 product license?", Some(false))? {
+            accept_license(ui)
+        } else {
+            ui.br()?;
+            ui.heading("+---------------------------------------------+")?;
+            Err(Error::LicenseNotAccepted)
+        }
+    }
+}
+
+pub fn accept_license(ui: &mut UI) -> Result<()> {
+    if license_exists() {
+        ui.info("You have already accepted the license.")?;
+        return Ok(());
+    }
+
+    ui.br()?;
+    ui.info("Accepting 1 product license...")?;
+
+    let license = LicenseData::new();
+    let content = serde_yaml::to_string(&license)?;
+    fs::create_dir_all(license_path())?;
+    let mut file = File::create(license_file())?;
+    file.write_all(content.as_bytes())?;
+
+    ui.status(ui::Status::Custom(ui::Glyph::CheckMark, String::from("")),
+              "1 product license accepted.")?;
+    ui.br()?;
+    ui.heading("+---------------------------------------------+")?;
+
+    Ok(())
+}
+
+fn license_path() -> PathBuf {
+    let hab_dir = if am_i_root() {
+        PathBuf::from(&*FS_ROOT_PATH).join("hab")
+    } else if let Some(home) = dirs::home_dir() {
+        home.join(".hab")
+    } else {
+        panic!("No home directory available. Unable to find a suitable place to write a license \
+                file.");
+    };
```

Comment:
This means that most times, a user will effectively have to accept the license twice, since there are some operations that can be run as non-root, and some that must be run as root.

There's also no indication of why the user is being asked to accept the license a second time, which could be confusing.

Reply:
This is the same pattern we use for other files, like `cli.toml`.  I'm happy to consider alternatives if you have any suggestions.